### PR TITLE
Multi TMC Support :)

### PR DESCRIPTION
**Pull request by Surion79**
Multi TMC Support, in a new PR

### CODE REVIEW

**Comment by Surion79** [comment]
Ok, VS Code created an automated PR with commit note. nice... Anyway, after messing up my first PR, here is the one

File: config/hardware/axis/XY/default_wiring.cfg [hunk 1]
```diff
@@ -4,14 +4,8 @@ dir_pin: X_DIR
...
-
 [stepper_y]
 step_pin: Y_STEP
 dir_pin: Y_DIR
 enable_pin: !Y_ENABLE
-endstop_pin: Y_STOP
-
-[tmc2209 stepper_y]
-uart_pin: Y_TMCUART
+endstop_pin: Y_STOP
```
Comment (Frix-x) [inline config/hardware/axis/XY/default_wiring.cfg:11]: Can you add a newline at the end of the file?

File: config/hardware/axis/Z/TriZero_tmc2209.cfg [hunk 2]
```diff
@@ -0,0 +1 @@
+[include ../tmc/z_tmc2209_1M.cfg]
```
Comment (Frix-x) [inline config/hardware/axis/Z/TriZero_tmc2209.cfg:1]: Can you add a newline at the end of the file?

File: config/hardware/axis/Z/Trident_tmc2209.cfg [hunk 3]
```diff
@@ -0,0 +1 @@
+[include ../tmc/z_tmc2209_3M.cfg]
```
Comment (Frix-x) [inline config/hardware/axis/Z/Trident_tmc2209.cfg:1]: Can you add a newline at the end of the file?

File: config/hardware/axis/Z/V0_tmc2209.cfg [hunk 4]
```diff
@@ -0,0 +1 @@
+[include ../tmc/z_tmc2209_1M.cfg]
```
Comment (Frix-x) [inline config/hardware/axis/Z/V0_tmc2209.cfg:1]: Can you add a newline at the end of the file?

File: config/hardware/axis/Z/V2.4_tmc2209.cfg [hunk 5]
```diff
@@ -0,0 +1 @@
+[include ../tmc/z_tmc2209_4M.cfg]
```
Comment (Frix-x) [inline config/hardware/axis/Z/V2.4_tmc2209.cfg:1]: Can you add a newline at the end of the file?

File: config/hardware/axis/Z/default_wiring_1M.cfg [hunk 6]
```diff
@@ -2,7 +2,4 @@
 step_pin: Z_STEP
 dir_pin: Z_DIR
 enable_pin: !Z_ENABLE
-endstop_pin: Z_STOP
-
-[tmc2209 stepper_z]
-uart_pin: Z_TMCUART
+endstop_pin: Z_STOP
```
Comment (Frix-x) [inline config/hardware/axis/Z/default_wiring_1M.cfg:5]: Can you add a newline at the end of the file?

File: config/hardware/axis/Z/default_wiring_3M.cfg [hunk 7]
```diff
@@ -7,13 +7,7 @@ step_pin: Z1_STEP
...
-uart_pin: Z1_TMCUART
-
 [stepper_z2]
 step_pin: Z2_STEP
 dir_pin: Z2_DIR
-enable_pin: !Z2_ENABLE
-
-[tmc2209 stepper_z2]
-uart_pin: Z2_TMCUART
+enable_pin: !Z2_ENABLE
```
Comment (Frix-x) [inline config/hardware/axis/Z/default_wiring_3M.cfg:13]: Can you add a newline at the end of the file?

File: config/hardware/axis/Z/default_wiring_4M.cfg [hunk 8]
```diff
@@ -5,7 +5,4 @@
 [stepper_z3]
 step_pin: Z3_STEP
 dir_pin: Z3_DIR
-enable_pin: !Z3_ENABLE
-
-[tmc2209 stepper_z3]
-uart_pin: Z3_TMCUART
+enable_pin: !Z3_ENABLE
```
Comment (Frix-x) [inline config/hardware/axis/Z/default_wiring_4M.cfg:8]: Can you add a newline at the end of the file?

File: config/hardware/axis/tmc/x_tmc2209.cfg [hunk 9]
```diff
@@ -0,0 +1,6 @@
+[tmc2209 stepper_x]
+uart_pin: X_TMCUART
+interpolate: False
+run_current: 0.8
+sense_resistor: 0.110
+stealthchop_threshold: 0
```
Comment (Frix-x) [inline config/hardware/axis/tmc/x_tmc2209.cfg:6]: Can you add a newline at the end of the file?

File: config/hardware/axis/tmc/y_tmc2209.cfg [hunk 10]
```diff
@@ -0,0 +1,6 @@
+[tmc2209 stepper_y]
+uart_pin: Y_TMCUART
+interpolate: False
+run_current: 0.8
+sense_resistor: 0.110
+stealthchop_threshold: 0
```
Comment (Frix-x) [inline config/hardware/axis/tmc/y_tmc2209.cfg:6]: Can you add a newline at the end of the file?

File: config/hardware/axis/tmc/y_tmc2209_V0specific.cfg [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+sense_resistor: 0.110
+# stealthchop_threshold: 999999
+driver_TBL: 2
+driver_TOFF: 2
+driver_HEND: 1
+driver_HSTRT: 4
+driver_PWM_LIM: 8 
+driver_PWM_GRAD: 8
+driver_PWM_FREQ: 1
+driver_PWM_REG: 4
```
Comment (Frix-x) [inline config/hardware/axis/tmc/y_tmc2209_V0specific.cfg:14]: Can you add a newline at the end of the file?

File: config/hardware/axis/tmc/z_tmc2209_1M.cfg [hunk 12]
```diff
@@ -0,0 +1,6 @@
+[tmc2209 stepper_z]
+uart_pin: Z_TMCUART
+interpolate: False
+run_current: 0.8
+sense_resistor: 0.110
+stealthchop_threshold: 0
```
Comment (Frix-x) [inline config/hardware/axis/tmc/z_tmc2209_1M.cfg:6]: Can you add a newline at the end of the file?

File: config/hardware/axis/tmc/z_tmc2209_3M.cfg [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+run_current: 0.8
+sense_resistor: 0.110
+stealthchop_threshold: 0
+
+[tmc2209 stepper_z2]
+uart_pin: Z2_TMCUART
+interpolate: False
+run_current: 0.8
+sense_resistor: 0.110
+stealthchop_threshold: 0
```
Comment (Frix-x) [inline config/hardware/axis/tmc/z_tmc2209_3M.cfg:15]: Can you add a newline at the end of the file?

File: config/hardware/axis/tmc/z_tmc2209_4M.cfg [hunk 14]
```diff
@@ -0,0 +1,8 @@
+[include z_tmc2209_3M.cfg]
+
+[tmc2209 stepper_z3]
+uart_pin: Z3_TMCUART
+interpolate: False
+run_current: 0.8
+sense_resistor: 0.110
+stealthchop_threshold: 0
```
Comment (Frix-x) [inline config/hardware/axis/tmc/z_tmc2209_4M.cfg:8]: Can you add a newline at the end of the file?

File: config/hardware/extruder/default_wiring.cfg [hunk 15]
```diff
@@ -3,7 +3,4 @@ step_pin: E_STEP
 dir_pin: E_DIR
 enable_pin: !E_ENABLE
 heater_pin: E_HEATER
-sensor_pin: E_TEMPERATURE
-
-[tmc2209 extruder]
-uart_pin: E_TMCUART
+sensor_pin: E_TEMPERATURE
```
Comment (Frix-x) [inline config/hardware/extruder/default_wiring.cfg:6]: Can you add a newline at the end of the file?

File: config/hardware/extruder/tmc/e_tmc2209.cfg [hunk 16]
```diff
@@ -0,0 +1,6 @@
+[tmc2209 extruder]
+uart_pin: E_TMCUART
+interpolate: false
+run_current: 0.45
+sense_resistor: 0.110
+stealthchop_threshold: 0
```
Comment (Frix-x) [inline config/hardware/extruder/tmc/e_tmc2209.cfg:6]: Can you add a newline at the end of the file?

File: config/hardware/extruder/tmc/e_tmc2240.cfg [hunk 17]
```diff
@@ -0,0 +1,8 @@
+[tmc2240 extruder]
+cs_pin: E_CS
+spi_software_sclk_pin: E_SCLK
+spi_software_mosi_pin: E_MOSI
+spi_software_miso_pin: E_MISO
+driver_TPFD: 0
+run_current: 0.650
+stealthchop_threshold: 0
```
Comment (Frix-x) [inline config/hardware/extruder/tmc/e_tmc2240.cfg:8]: Can you add a newline at the end of the file?

**Review by Frix-x — CHANGES_REQUESTED** [review]
So I have done a first pass on the PR.
This looks promising but there is multiple files without the newline at the end of the file (I think I will try to find a way to enforce it automatically using commit hooks).

Also it look like it's not finished as there is nothing in the printer.cfg at the moment.

**Comment by Surion79** [comment]
Yeah, unfortunately i scrapped by accident my initial PR. Printer.cfg was already prepped

**Comment by Frix-x** [comment]
I merged the PR into a branch in my repo to be able to work on it correctly

**Pull request merged**